app/src/main/java/com/mealsmadeeasy/ui/home/discover/SearchActivity.kt

```diff
@@ -0,0 +1,89 @@
+package com.mealsmadeeasy.ui.home.discover
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.support.v7.widget.LinearLayoutManager
+import android.support.v7.widget.RecyclerView
+import android.view.Menu
+import com.mealsmadeeasy.MealsApplication
+import com.mealsmadeeasy.R
+import com.mealsmadeeasy.data.MealStore
+import com.mealsmadeeasy.ui.BaseActivity
+import javax.inject.Inject
+import android.support.v7.widget.SearchView
+import android.util.Log
+import android.widget.Toast
+import com.mealsmadeeasy.ui.meal.MealActivity
+import io.reactivex.Observable
+import io.reactivex.subjects.BehaviorSubject
+import java.util.concurrent.TimeUnit
+
+
+class SearchActivity : BaseActivity() {
+
+    @Inject lateinit var mealStore: MealStore
+    private var querySubject : BehaviorSubject<String> = BehaviorSubject.create()
+
+    companion object {
+        private const val TAG = "SearchActivity"
+        fun newIntent(context: Context?) = Intent(context, SearchActivity::class.java)
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        MealsApplication.component(this).inject(this)
+
+        setContentView(R.layout.activity_search)
+
+        val toolbar = findViewById<android.support.v7.widget.Toolbar>(R.id.search_toolbar)
+        setSupportActionBar(toolbar)
+
+        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        supportActionBar?.setDisplayShowHomeEnabled(true)
+
+        val searchResults = findViewById<RecyclerView>(R.id.search_results_list)
+
+        getQueryObservable()
+                .debounce(2, TimeUnit.SECONDS)
```

Comment:
2 seconds feels a bit long here. I played around with it and I think 500 milliseconds feels better here.